talktorials/11_online_apis/11a_obtaining_structures.ipynb

```diff
@@ -0,0 +1,2943 @@
+    "\n",
+    "### PubChem\n",
+    "\n",
+    "* Role: Small compounds database\n",
+    "* Website: https://pubchem.ncbi.nlm.nih.gov/\n",
+    "* API: Yes, REST-based. No official client. Use `bravado`.\n",
+    "* Documentation: https://pubchemdocs.ncbi.nlm.nih.gov/pug-rest\n",
+    "* Literature:\n",
+    "    * Nucleic Acids Res. 2019 Jan 8; 47(D1):D1102-1109. https://academic.oup.com/nar/article/47/D1/D1102/5146201\n",
+    "\n",
+    "> PubChem is an open chemistry database at the National Institutes of Health (NIH). “Open” means that you can put your scientific data in PubChem and that others may use it. Since the launch in 2004, PubChem has become a key chemical information resource for scientists, students, and the general public. Each month our website and programmatic services provide data to several million users worldwide.\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "***\n",
+    "\n",
+    "# Practical\n",
+    "\n",
+    "## Building the pipeline\n",
+    "\n",
+    "### Obtaining information from KLIFS\n",
+    "\n",
+    "Choose a random kinase from the EGFR family (in `mol2`) and the corresponding ligands (`SMILES`) from the KLIFS database. We will also obtain the PDB structure of the protein-ligand complex so we can calculate the binding pockets in part B."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from bravado.client import SwaggerClient\n",
+    "KLIFS_API_DEFINITIONS = \"http://klifs.vu-compmedchem.nl/swagger/swagger.json\"\n",
+    "KLIFS_CLIENT = SwaggerClient.from_url(KLIFS_API_DEFINITIONS, config={'validate_responses': False})"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def _all_kinase_families():\n",
+    "    return KLIFS_CLIENT.Information.get_kinase_families().response().result\n",
+    "\n",
+    "def _kinases_from_family(family, species=\"HUMAN\"):\n",
+    "    return KLIFS_CLIENT.Information.get_kinase_names(kinase_family=family, species=species).response().result\n",
+    "\n",
+    "def _protein_and_ligand_structure(*kinase_ids):\n",
+    "    structures = KLIFS_CLIENT.Structures.get_structures_list(kinase_ID=kinase_ids).response().result\n",
+    "    molcomplex = KLIFS_CLIENT.Structures.get_structure_get_pdb_complex(structure_ID=structures[0].structure_ID).response().result\n",
+    "    protein = KLIFS_CLIENT.Structures.get_structure_get_protein(structure_ID=structures[0].structure_ID).response().result\n",
+    "    ligands = KLIFS_CLIENT.Ligands.get_ligands_list(kinase_ID=kinase_ids).response().result\n",
+    "    return molcomplex, protein, [ligand.SMILES for ligand in ligands]"
```

Comment:
Maybe add a print statement that tells the user which KLIFS structure is selected (and visualized later)?

` print(f"Chosen KLIFS entry: PDB ID {structures[0].pdb} with chain {structures[0].chain} and alternate model {structures[0].alt}")`

Reply:
We have one at the `step_01*` functions, I think!

Reply:
Nvm, I just saw what you mean. Updated.